### DISCUSSION TITLE
Getting an error when trying to view or edit printer profiles

### DISCUSSION THREAD

**Discussion by EricTSten** [OctoFarm/OctoFarm#927] [General]
This just started about a week ago.  When I try to view or edit a printer profile I get the following error:

"Something has gone wrong with loading the Printer Settings... Hard Failure, please submit as a bug on github: TypeError: p.currentProfile is undefined" 

![image](https://user-images.githubusercontent.com/87784951/168080206-522391ba-3fa7-45af-9216-be4af02e1140.png)

So, I'm doing as requested, submitting an issue. :-) 

Suggestions on what to do now?

**Comment by EricTSten** [comment]
I meant to add to my original post.  I've restarted the OctoFarm server, both software as well as OS without any change to the error message.

**Reply by NotExpectedYet** [reply]
You can try a API rescan to collect the information but that would happen on a restart anyway so doubt it would resolve the issue. Seems like it's something specifically with collecting the data from OP. Dunno why it's just started now but some logs would help further, and an OctoFarm version... 

Hopefully I've resolved it for V1.2 but it would be good to know if your comfortable with git I can pass instructions for you to try the latest RC. It's stable, just check the issues listed in #912.

**Reply by EricTSten** [reply]
Yeah, the version info would have been helpful from the start, I apologize for that. :-)  I am running version 1.1.13-hotfix.  

I'm willing to give version 1.2 a try.  :-) 

  Thanks! 

      Eric  

[octofarm_dump.zip](https://github.com/OctoFarm/OctoFarm/files/8688501/octofarm_dump.zip)

**Reply by NotExpectedYet** [reply]
No trouble at all! 

I've had a look through and yeah for some strange reason the settings api endpoint (the one that contains the information on the profile) is taking a very very long time to respond. There's no other issue that I can see other than a timeout. 

OctoFarm isn't dealing with the lack of that information very well, but I'm wondering if the upgrade would actually help here. Could try setting your timeout settings up to the max they will allow and see if that allows it to grab the information? They're in System -> Server -> Timeout. The two API one's should be all that's required.

The new version will help a little if it manages to grab that information in the first place. It saves it now, rather than re-grabbing every restart but need to see why it's taking so long first or will not help at all. 

05/05/2022, 19:43:29 | ERROR | OctoFarm-State | Error attempting to connect to API: api/settings | http://ender3pro.deamon.local | timeout: 1000data:"\"timeout\"" |

**Reply by NotExpectedYet** [reply]
One thing to try would be connecting over the local ip rather than mDNS. I do see lot's of weird issues with that method of connection and OctoFarm doesn't seem to deal with it very well. Doesn't explain your issue if it fixes it but worth a try. If those don't work, then I'll write some upgrade instructions for you to force the update to the development branch.

**Reply by EricTSten** [reply]
I've adjusted the timeout to 10 seconds with no change.  I actually only connect via the local IP, not the mDNS entry.  One other thing I'm finding is that my printers don't show up under the panel or list view since this error started appearing.  

Thanks

   Eric

**Reply by NotExpectedYet** [reply]
Yeah checked further back in the logs, but was still the same with IP's... 

There was an issue with displaying Offline/Online printers but there was also a bug where they would fail to load without that profile information. 

May as well try the newer version

`git fetch origin development && git checkout development && git pull`
Run that inside the OctoFarm/ directory
Then install the new dependencies
`cd server/ && npm ci`
Then restart the pm2 service
`pm2 restart OctoFarm`
Should be updated on boot
Should see RC10 on System page for server version

**Reply by NotExpectedYet** [reply]
I'm currently down to 0 bugs again after finding a few in RC9 this morning. 

So hopefully it has sorted your issue as well in the various changes that have gone on...

**Reply by EricTSten** [reply]
I'm not too familiar with pulling files from git so I don't know if this is "normal" or not but I got an error after running the 1st command:
<img width="834" alt="Screen Shot 2022-05-17 at 8 37 35 AM" src="https://user-images.githubusercontent.com/87784951/168812841-15af0572-7032-4c0b-acca-639a7ab5ecad.png">

It seems like the _git checkout development_ command is where it failed.  I was able to manually run _git pull_ and it looks like it pulled more files. 

However, I'm guessing something didn't work properly since there is no server directory under Octofarm to run the commands to install the new dependencies.  

Suggestions?

  Thank you

     Eric

**Reply by NotExpectedYet** [reply]
Ah the depth, good spot. 

`git remote update origin` 

Then `git checkout development`

I "think" should work. Sorry forgot the new instructions don't pull all the branches by default. 

Let me know if it doesn't and I'll advise further.

**Reply by NotExpectedYet** [reply]
The `cd server && npm ci` command may take a little longer than usual as the client is built locally now with a different package manager just to warn.

**Reply by EricTSten** [reply]
I think I must still be doing something wrong here. :-)  I ran the `git remote update origin` command and it looks like that grabbed a bunch of stuff:

<img width="624" alt="image" src="https://user-images.githubusercontent.com/87784951/168878649-afc4b8b8-1b3f-441b-b1aa-9c4b1ebee2b2.png">

And then I tried the `git checkout development` command and got the same error as before.  
<img width="559" alt="image" src="https://user-images.githubusercontent.com/87784951/168879271-2470966d-240d-4cef-afe1-5f2d44daf366.png">

I'm going to assume that the server directory is created at some point when the correct files are pulled down because my installation still does not have an OctoFarm/servers directory.

  Thanks

      Eric Sten

**Reply by NotExpectedYet** [reply]
Yeah sorry your correct, the server dir is the new server_src and will be available when I figure that out 😅. I'll do some investigation and get back to you. Hopefully I've not dug a hole with depth 0 but the full 1.2 will be out this weekend if I done a goof with that.

**Reply by NotExpectedYet** [reply]
`git remote set-branches origin '*'`
`git fetch -v`
`git checkout development`
`cd server && npm ci`
`pm2 restart OctoFarm`

And hopefully nothing blows up. 

Sorry posted twice didn't reply with the other

**Comment by NotExpectedYet** [comment]
`git remote set-branches origin '*'`
`git fetch -v`
`git checkout development`
`cd server && npm ci`
`pm2 restart OctoFarm`

And hopefully nothing blows up.

**Reply by EricTSten** [reply]
Well, there has been progress, the error message has changed. :-)   I am able to checkout the development branch now but when I try to build it with `npm ci` I get this error
`npm ERR! fsevents not accessible from chokidar`

A quick google search lead me to updating NPM which moved past the error and on to the build/install process, which completed but with some warning about assets exceeding the recommended size limit of 244KiB.

I restarted OctoFarm via PM2 and now it is not accessible.

**Reply by EricTSten** [reply]
I checked the pm2 logs and found that it can't find the app.js module.  It is looking for it in the /home/sysadmin/OctoFarm directory when it actually is in /home/sysadmin/OctoFarm/server directory.

**Reply by NotExpectedYet** [reply]
Something sounds weird there, let me double check this end and make sure nothings gone weird. Will get back to you.

**Reply by NotExpectedYet** [reply]
It's pm2 service holding onto the old file... grr.. aha. 

From the OctoFarm directory, not server

`git pull` - I found a little issue but just resolved. 

`pm2 delete OctoFarm`

`npm start` 

Will resolve. 

You'll just need to re-run the startup stuff again here: https://docs.octofarm.net/installation/setup-service.html

Glad we ran into those because that last one will break my auto updater process... for the live release.

**Reply by EricTSten** [reply]
Success!  

Version 1.2.0 is now up and running and I'm able to see the printers in the different views.  The original profile error message has gone away.  

Thank you for all your help!  

   Eric Sten

**Reply by EricTSten** [reply]
Well, almost no issues.  I got this error the first time I went to the dashboard but I thought it was a one time thing since it didn't show up again.  Well, I was wrong.

Should I open a new issue for this since it isn't related to the printer profiles?

TypeError ERROR (undefined):
Please report this error to [OctoFarm Issues](https://github.com/octofarm/octofarm/issues)!
t[0] is null
FILE INFO
LINE: 2
COL: 33821
FILE:
http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js
STACK:
value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:33821 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:482260 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:5949 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:488159 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:488087 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:385883 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:385542 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:487274 ct@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:635695 updateCurrentUtilisation@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:637115 currentStatusAndUtilisation@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:639297 @http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:665550 ft/ut.onmessage@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:637419

**Reply by NotExpectedYet** [reply]
> Well, almost no issues. I got this error the first time I went to the dashboard but I thought it was a one time thing since it didn't show up again. Well, I was wrong.
> 
> Should I open a new issue for this since it isn't related to the printer profiles?
> 
> TypeError ERROR (undefined): Please report this error to [OctoFarm Issues](https://github.com/octofarm/octofarm/issues)! t[0] is null FILE INFO LINE: 2 COL: 33821 FILE: http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js STACK: value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:33821 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:482260 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:5949 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:488159 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:488087 value/<@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:385883 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:385542 value@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:487274 ct@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:635695 updateCurrentUtilisation@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:637115 currentStatusAndUtilisation@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:639297 @http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:665550 ft/ut.onmessage@http://192.168.0.25:4000/assets/dist/dashboard.runner.min.js:2:637419

Yeah log an issue totally. 

I have been chasing that myself but could never reproduce it. 

Could you do a full log dump that's in the system page and add it too the issue please? I "think" it might be related to your original issue and I'm possibly hiding it better. I could only reproduce that issue when restarting the server and having the dashboard open so i chalked it up to a missing bit of info but I'll take a deeper look.

**Reply by NotExpectedYet** [reply]
One more thing sorry aha. 

Just open the health checks page on printer manager before doing the log dump. Will help narrow it down :) 

Thanks for trying this btw!
